src/Microsoft.DotNet.Darc/src/DarcLib/Helpers/HttpRequestManager.cs

```diff
@@ -54,6 +55,17 @@ public async Task<HttpResponseMessage> ExecuteAsync(int retryCount = 15)
                         }
 
                         HttpResponseMessage response = await _client.SendAsync(message);
+
+                        if (response.StatusCode == HttpStatusCode.NotFound)
+                        {
+                            if (_logFailure)
+                            {
+                                _logger.LogError("A 404 (Not Found) was returned. We'll set the retries amount to 0.");
+                            }
+
+                            retriesRemaining = 0;
+                        }
+
                         response.EnsureSuccessStatusCode();
```

Comment:
This will still throw on 404.  Should we just return the response here?

Reply:
Yeah, it will throw and the exception will be caught bellow but it won't retry. I did this since some methods are expecting a 404 so they can do something else so we continue to do so without retrying

Reply:
oh oh, they catch the exception.